### PR TITLE
Enhancement/refactor

### PR DESCRIPTION
**Pull request by tieswesterborg**
Refactored most of the classes, Also fixed some remaining bugs.

### CODE REVIEW

File: src/test/java/core/graph/PhylogeneticTreeTest.java [hunk 1]
```diff
@@ -15,62 +12,54 @@
...
-    public void testSetup() {
-        pt.setModel(mockedModel);
-        try {
-            pt.setup();
-            verify(mockedModel, atLeast(1)).addCell(anyInt(), anyString(), any(CellType.class));
-        } catch (IOException e) {
-            e.printStackTrace();
-        }
-    }
+//    private PhylogeneticTree pt;
```
Comment (ghost) [inline src/test/java/core/graph/PhylogeneticTreeTest.java:15]: Failing test cases are commented. If you make them fail, fix it.

**Comment by ghost** [comment]
CheckStyle, PMD and FindBugs violations are still present, before a PR all of them should be resolved.

File: src/main/java/application/controllers/GraphController.java [hunk 2]
```diff
@@ -98,42 +81,33 @@ public void initialize(URL location, ResourceBundle resources) {
...
-        List<Cell> list = graph.getModel().getAddedCells();
-
-        for (Cell c : list) {
-            graphMouseHandling.setMouseHandling(c);
-        }
-
-        graph.endUpdate();
-
-
+        /**
```
Comment (nubcaker) [inline src/main/java/application/controllers/GraphController.java]: JavaDoc should not be present inside a method.

**Comment by arbreurkes** [comment]
Looking good!

**Pull request merged**